akvo/rsr/management/commands/single_period_indicators.py

```diff
@@ -40,6 +40,23 @@ def get_parent_periods(indicator, PERIOD_START, PERIOD_END):
                 period_start=PERIOD_START, period_end=PERIOD_END
             )
 
+        def find_periods_with_data(periods):
+            """ Separate a list of periods into two groups: those tht have been modified from an
+            original, "empty" state and those that haven't. To be counted as modified a period has
+            data in at least one of the fields target_value, target_comment, actual_value or
+            actual_comment or has at least one update.
+            """
+            modified_periods = []
+            unmodified_periods = []
+            for period in periods:
+                updates = IndicatorPeriodData.objects.filter(period=period)
```

Comment:
I would change this to check for `updates.exists()` since `if updates` is much slower than `if updates.exists()`